includes/polling/applications/bird2.inc.php

```diff
@@ -0,0 +1,192 @@
+    ];
+
+    // Deal with the rest of the body
+    $protocolBodys = preg_split("/^\s{2}([A-Z])/m", $protocolSegmentParts[1]);
+
+    // Loop through all BGP protocols
+    foreach ($protocolBodys as $protocolBody) {
+        // Deal with the BGP block
+        if (strpos($protocolBody, 'GP') === 0) {
+            foreach (explode("\n", 'B' . $protocolBody) as $protocolBodyLine) {
+                if (strpos($protocolBodyLine, ':') !== false) {
+                    $lineParts = explode(':', $protocolBodyLine, 2);
+                    $protocolData[str_replace(' ', '_', strtolower(trim($lineParts[0])))] = trim($lineParts[1]);
+                }
+            }
+
+            // Fix up the error string
+            if (isset($protocolData['last_error'])) {
+                // Trim the received
+                $protocolData['last_error'] = trim(str_ireplace('Received:', '', $protocolData['last_error']));
+            }
+        }
+
+        // Process the Ip channel (v4/v6)
+        $IpVersion = 4;
+        if (filter_var($protocolData['neighbor_address'], FILTER_VALIDATE_IP, FILTER_FLAG_IPV6)) {
+            $IpVersion = 6;
+        }
+
+        if (strpos($protocolBody, 'hannel ipv' . $IpVersion) === 0) {
+
+            foreach (explode("\n", 'C' . $protocolBody) as $protocolBodyLine) {
+                if (strpos($protocolBodyLine, ':') !== false) {
+                    $lineParts = explode(':', $protocolBodyLine, 2);
+                    $protocolData[str_replace(' ', '_', strtolower(trim($lineParts[0])))] = trim($lineParts[1]);
+                }
+            }
+
+            // Fix up the ROUTES
+            if (isset($protocolData['routes'])) {
+                $routeParts = explode(', ', $protocolData['routes']);
+                unset($protocolData['routes']);
+                foreach ($routeParts as $routePart) {
+                    $routeDetail = explode(' ', $routePart);
+                    $protocolData['routes'][$routeDetail[1]] = $routeDetail[0];
+                }
+            }
+
+            // Set the route updates
+            unset($protocolData['route_change_stats']);
+            foreach (['import_updates', 'import_withdraws', 'export_updates', 'export_withdraws'] as $key) {
+                if (!isset($protocolData[$key])) {
+                    continue;
+                }
+
+                $routeChange_parts = preg_split("/\s+/", trim($protocolData[$key]));
+
+                unset($protocolData[$key]);
+                $protocolData['route_change_stats'][$key] = [
+                    'received' => $routeChange_parts[0],
+                    'rejected' => $routeChange_parts[1],
+                    'filtered' => $routeChange_parts[2],
+                    'ignored' => $routeChange_parts[3],
+                    'accepted' => $routeChange_parts[4],
+                ];
+            }
+        }
+    }
+
+    $protocolsData[] = $protocolData;
+}
+
+// ---
+$deviceObj = Device::findOrFail($device['device_id']);
```

Comment:
```suggestion
$deviceObj = DeviceCache::getPrimary();
```
Object already exists, avoid fetching it from the database.